app/models/tariff_update.rb

```diff
@@ -4,62 +4,48 @@ class TariffUpdate
 
   collection_path '/admin/updates'
 
-  attributes :update_type, :state, :issue_date, :created_at, :updated_at, :applied_at, :filesize,
-             :exception_backtrace, :exception_class, :exception_queries,
-             :file_presigned_url, :log_presigned_urls, :presence_errors
+  attributes :update_type,
+             :state,
+             :issue_date,
+             :created_at,
+             :updated_at,
+             :applied_at,
+             :filesize,
+             :exception_backtrace,
+             :exception_class,
+             :exception_queries,
+             :file_presigned_url,
+             :presence_errors
 
   STATES = {
     'A' => 'Applied',
-    'M' => 'Missing',
     'P' => 'Pending',
     'F' => 'Failed',
   }.freeze
 
+  ROLLBACK_APPLICABLE_STATES = %w[A P].freeze
```

Comment:
Surely we only want to allow the user to rollback to an update which has been applied?

Reply:
Yeah, that makes sense to me. I preserved the existing logic, BTW